src/charm.py

```diff
@@ -63,12 +82,104 @@ def _on_website_relation_joined(self, event):
             'port': str(port)
         })
 
+    def _setup_promtail(self):
+        """Sets up the promtail service on the machine. We don't have to worry
+        about removal of Promtail since you can't remove the controller charm.
+        """
+        # Download promtail binary
+        check_call(["curl", "-LO", "https://github.com/grafana/loki/releases/download/v2.6.1/promtail-linux-amd64.zip"])
```

Comment:
Not sure we can leave this in the controller charm - how will this work in airgapped environments?